launch_testing/launch_testing/__init__.py

```diff
@@ -53,14 +53,15 @@ def __init__(
         self.name = name
         self.launch_descriptor = launch_descriptor
         self.expected_lines = expected_lines
-        self.expected_output = b'\n'.join(self.expected_lines)
+        self.expected_output = b'\n'.join(self.expected_lines) + b'\n'
         self.left_over_stdout = b''
         self.left_over_stderr = b''
         self.stdout_data = io.BytesIO()
         self.stderr_data = io.BytesIO()
         self.regex_match = regex_match
         self.exit_on_match = exit_on_match
         self.matched = False
+        self.matched_exactly = False
```

Comment:
thanks - 
the tests written with this in mind didn't catch this because all of the output lines were being received in the one callback. Is https://github.com/ros2/launch/pull/34/commits/4167c2eb831d05fe82a33a1683255a4c2e2567fb an appropriate way to address this? (it works but you might have better ideas)

I have removed [these lines](https://github.com/ros2/launch/pull/34/files#diff-95fa932db2912f5a4060524ea72cd0cfL66). There are some tests that start failing in other places e.g. talker-listener in `rclcpp_examples` as a result, but I'll adapt them (assuming you agree that requiring `matched_exactly` should be the default behaviour)